src/tic-tac-toe.ts

```diff
@@ -40,35 +43,70 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinningVector | undefined {
+    for (let rowIndex = 0; rowIndex < this.board.length; rowIndex++) {
+      const row = this.board[rowIndex];
+      const vector: WinningVector = [];
+      for (let colIndex = 0; colIndex < this.board.length; colIndex++) {
+        const square = row[colIndex];
+        if (this.isCurrentPlayer(square)) {
+          vector.push({ row: rowIndex, col: colIndex });
+        }
+      }
+      if (vector.length === this.boardSize) {
+        return vector;
+      }
+    }
+    return;
   }
 
-  private checkColumnsForWin(): boolean {
+  private checkColumnsForWin(): WinningVector | undefined {
     const transposedBoard = this.eachIndex().map(i =>
       this.board.map(row => row[i])
     );
 
-    return transposedBoard.some(row => row.every(this.isCurrentPlayer));
+    for (let colIndex = 0; colIndex < transposedBoard.length; colIndex++) {
+      const column = transposedBoard[colIndex];
+      const vector: WinningVector = [];
+
+      for (let rowIndex = 0; rowIndex < transposedBoard.length; rowIndex++) {
+        const square = column[rowIndex];
+        if (this.isCurrentPlayer(square)) {
+          vector.push({ row: rowIndex, col: colIndex });
+        }
+      }
+      if (vector.length === this.boardSize) {
+        return vector;
+      }
+    }
+    return;
   }
 
-  private checkDiagonalsForWin(): boolean {
-    const diagonals = [
-      this.eachIndex().map(i => this.board[i][i]),
-      this.eachIndex().map(i => this.board[i][this.boardSize - 1 - i]),
-    ];
+  private checkDiagonalsForWin(): WinningVector | undefined {
+    const winningVector: WinningVector = [];
 
-    return diagonals.some(values => values.every(this.isCurrentPlayer));
+    for (let i = 0; i < this.boardSize; i++) {
+      if (this.isCurrentPlayer(this.board[i][i])) {
+        winningVector.push({ row: i, col: i });
+      } else if (this.isCurrentPlayer(this.board[i][this.boardSize - 1 - i])) {
+        winningVector.push({ row: i, col: this.boardSize - 1 - i });
+      }
+      if (winningVector.length === this.boardSize) {
+        return winningVector;
+      }
+    }
+    return;
   }
 
   private checkGameForWin(): void {
-    const winCondition = [
+    const winVectors = [
       this.checkRowsForWin(),
       this.checkColumnsForWin(),
-      this.checkDiagonalsForWin()
+      this.checkDiagonalsForWin(),
     ];
 
-    if (winCondition.some(winnerCheck => winnerCheck)) {
+    if (winVectors.some(vector => !!vector)) {
```

Comment:
What's happening in there? Here's how I read it:
> "If in the set of winning vectors we find any that aren't false, then set the winning vector to be the one that isn't false or an empty array".

The fact that you are checking for any winning vectors -- fair! -- and then on the next line finding that winning vector *or* giving a default is a smell to me. 

I'd suggest: 
```suggestion
    const maybeWinningVector = winVectors.find(vector => !!vector);
    if (maybeWinningVector)) {
```

If nothing is found in the call to `find`, you get `undefined` -- so you won't hop through that logic gate then. You can then skip the step where you call `find` below.

I definitely wouldn't assign `[]` if no winner is found -- not only should that *never* happen (and if it does as written, something has gone horrifically wrong, and things should crash horrifically downstream if it were to), it's also false, unless `[]` is actually a valid win condition.

Reply:
Addressed in 8daf3c05986377c78cd5e0c701ad941a83721ad0